service/installer/helm/helm.go

```diff
@@ -0,0 +1,355 @@
+	}
+}
+
+// Installer is an implementation of the Installer interface, that uses Helm to
+// install charts.
+type Installer struct {
+	// Dependencies.
+	configurers []configurerspec.Configurer
+	fileSystem  afero.Fs
+	logger      micrologger.Logger
+
+	// Settings.
+	helmBinaryPath string
+	organisation   string
+	password       string
+	registry       string
+	username       string
+}
+
+// New creates a new configured Helm Installer.
+func New(config Config) (*Installer, error) {
+	// Dependencies.
+	if config.Configurers == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Configurers must not be empty")
+	}
+	if config.FileSystem == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.FileSystem must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	// Settings.
+	if config.HelmBinaryPath == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.HelmBinaryPath must not be empty")
+	}
+	if config.Organisation == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Organisation must not be empty")
+	}
+	if config.Password == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Password must not be empty")
+	}
+	if config.Registry == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Registry must not be empty")
+	}
+	if config.Username == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Username must not be empty")
+	}
+
+	exists, err := afero.Exists(config.FileSystem, config.HelmBinaryPath)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if !exists {
+		return nil, microerror.Maskf(invalidConfigError, "helm binary path '%s' does not exist", config.HelmBinaryPath)
+	}
+
+	installer := &Installer{
+		// Dependencies.
+		configurers: config.Configurers,
+		fileSystem:  config.FileSystem,
+		logger:      config.Logger,
+
+		// Settings.
+		helmBinaryPath: config.HelmBinaryPath,
+		organisation:   config.Organisation,
+		password:       config.Password,
+		registry:       config.Registry,
+		username:       config.Username,
+	}
+
+	if err := installer.login(); err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return installer, nil
+}
+
+func (i *Installer) Install(project spec.Project) error {
+	i.logger.Log("debug", "ensuring chart is installed", "name", project.Name, "ref", project.Ref)
+
+	var err error
+
+	// We create a tmp dir in which all Helm values files and tarballs are written
+	// to. After we are done we can just remove the whole tmp dir to clean up.
+	var tmpDir string
+	{
+		tmpDir, err = afero.TempDir(i.fileSystem, "", "draughtsman-operator-helm-installer")
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		defer func() {
+			err := i.fileSystem.RemoveAll(tmpDir)
+			if err != nil {
+				i.logger.Log("error", fmt.Sprintf("could not remove tmp dir: %#v", err), "name", project.Name, "ref", project.Ref)
+			}
+		}()
+	}
```

Comment:
This is a bit restructured. The tarball was downloaded to "somewhere". Now everything is in the tmpdir which is removed afterwards.